src/pages/Hosting/HostingRouter.js

```diff
@@ -1,52 +1,96 @@
 import React from 'react';
 import { Route, Routes, useNavigate, useLocation } from 'react-router-dom';
+import styled from 'styled-components';
 import HostingFooter from './HostingComponents/HostingFooter';
 import HostingNav from './HostingComponents/HostingNav';
+import StayType from './HostingPages/StayType';
 import SearchForm from './HostingPages/Location/SearchForm';
-import styled from 'styled-components';
-import PrivacyType from './HostingPages/StayType';
+import FloorPlan from './HostingPages/FloorPlan/FloorPlan';
 
-const Customer = () => {
+const HostingRouter = () => {
   const navigate = useNavigate();
   const location = useLocation();
   const newStayInfo = {};
-  let mainQuestion = '';
 
-  //TODO : 추후 switch 문으로 변경예정
-  if (location.pathname === '/hosting') {
-    mainQuestion = '게스트가 머무르게 될 숙소의 종류가 무엇인가요?';
-  } else if (location.pathname === '/hosting/location') {
-    mainQuestion = '숙소 위치는 어디인가요?';
-  }
+  location.pathname === '/hosting' && localStorage.removeItem('stayInfo');
+
+  const addInfo = newStayInfo => {
+    let info = [];
+    if (localStorage.getItem('stayInfo')) {
+      info = JSON.parse(localStorage.getItem('stayInfo'));
+    }
+    info.push(newStayInfo);
+    localStorage.setItem('stayInfo', JSON.stringify(info));
+  };
+
+  const removeInfo = () => {
+    let storageInfo = JSON.parse(localStorage.getItem('stayInfo'));
+    storageInfo.pop();
+    localStorage.setItem('stayInfo', JSON.stringify(storageInfo));
+  };
+
+  const getPageTitle = path => {
+    const title = {
+      '/hosting': '게스트가 머무르게 될 숙소의 종류가 무엇인가요?',
+      '/hosting/location': '숙소 위치는 어디인가요?',
+      '/hosting/floor-plan': '숙소에서 맞이할 최대 인원수를 알려주세요',
+      '/hosting/amenities': '숙소 편의시설 정보를 추가해 주세요',
+    };
+    return title[path];
+  };
 
-  //TODO : 추후 Switch 문으로 변경예정
   const goToNextStep = () => {
-    if (location.pathname === '/hosting') {
-      navigate('/hosting/location');
+    addInfo(newStayInfo);
+    switch (location.pathname) {
+      case '/hosting':
+        navigate('/hosting/location');
+        break;
+      case '/hosting/location':
+        navigate('/hosting/floor-plan');
+        break;
+      case '/hosting/floor-plan':
+        navigate('/hosting/amenities');
+        break;
+      default:
+        navigate('/hosting');
     }
   };
 
-  //TODO : 추후 switch 문으로 변경예정
   const goToPrevStep = () => {
-    if (location.pathname === '/hosting/location') {
-      navigate('/hosting');
+    removeInfo();
+    switch (location.pathname) {
+      case '/hosting/location':
+        navigate('/hosting');
+        break;
+      case '/hosting/floor-plan':
+        navigate('/hosting/location');
+        break;
+      case '/hosting/amenities':
+        navigate('/hosting/floor-plan');
+        break;
+      default:
+        navigate('/hosting');
     }
```

Comment:
단순히 이전페이지로 가는거면 역산할 필요없이 
navigate(-1)로 해주시면 됩니다!
```suggestion
navigate(-1)
```

Reply:
와...이거 진짜 신세계네요 😮 몇 줄의 코드를 저 하나로 줄일 수 있다니...! 변경해 주었습니다! 감사합니다.